cmd/m3/dev.go

```diff
@@ -0,0 +1,260 @@
+		}
+	}()
+	config := &rest.Config{
+		// TODO: switch to using cluster DNS.
+		Host:            "http://localhost:8001",
+		TLSClientConfig: rest.TLSClientConfig{},
+		BearerToken:     "eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJkZWZhdWx0Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZWNyZXQubmFtZSI6ImRhc2hib2FyZC10b2tlbi1mZ2J4NSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJkYXNoYm9hcmQiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC51aWQiOiIyNGE3Mjg1OC00YjE4LTRhZDEtYjM4YS03ZTA2NGM2ODI1ZmEiLCJzdWIiOiJzeXN0ZW06c2VydmljZWFjY291bnQ6ZGVmYXVsdDpkYXNoYm9hcmQifQ.OTj-gB3OnDA5yDmtRZVF9wxMx-6fT1o3vSmd_lZrCpddTBgSkUb2vnaB8eVDQ_DKN2fHsnWw6JvZoPftJ27gKVZ_dAM_21XwgUJy72_lhI_XLinGcx5TAqObxhLp5-YlCTQPDbVEW56DUs59mvx2KKaYeeS7KE-ORYN4wpH6ecZnhUR7_jhSdJAb9MBp3reUU6Iou2YDfEHtHgrSoF7EpZrQME8zjtTQE0Fkl6YavKA1zjHMg-yKuiFRjLkKcrcXyYa_j4lFXL_ZGEICy94FsjGAPv4iwCqZW9ruTU9EX0B0BbG4xGYEZfgG6B5iqIUdleYzHl86eSpWQMS5H5xguQ",
+		BearerTokenFile: "some/file",
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Println("ERROR WITH INFORMER")
+	}
+
+	publicCh := servicePortForwardPort(m3PFCtx, kindk8sConf, "m3", "50051", color.FgYellow)
+	privateCh := servicePortForwardPort(m3PFCtx, kindk8sConf, "m3", "50052", color.FgGreen)
+	nginxCh := servicePortForwardPort(nginxCtx, kindk8sConf, "nginx-resolver", "9000:80", color.FgCyan)
+	initialized := false
+	nginxInitialized := false
+
+	// informer factory
+	factory := informers.NewSharedInformerFactory(clientset, 0)
+
+	// monitor m3 with pod informer
+	podInformer := factory.Core().V1().Pods().Informer()
+	podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*v1.Pod)
+			if strings.HasPrefix(pod.ObjectMeta.Name, "m3") {
+				fmt.Println("An m3 pod was added:", pod.ObjectMeta.Name)
+				// we are going to ignore the first registered pod
+				if !initialized {
+					initialized = true
+					fmt.Println("Initialized Pod Informer")
+				} else {
+					// close private and public so they get restarted
+					m3Cancel()
+					// restart the context
+					m3PFCtx, m3Cancel = context.WithCancel(context.Background())
+				}
+			}
+			// monitor nginx
+			if strings.HasPrefix(pod.ObjectMeta.Name, "nginx") {
+				fmt.Println("An nginx pod was added:", pod.ObjectMeta.Name)
+				// we are going to ignore the first registered pod
+				if !nginxInitialized {
+					nginxInitialized = true
+					fmt.Println("Initialized Pod Informer")
+				} else {
+					// close private and public so they get restarted
+					nCancel()
+					// restart the context
+					nginxCtx, nCancel = context.WithCancel(context.Background())
+				}
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			pod := obj.(*v1.Pod)
+			if strings.HasPrefix(pod.ObjectMeta.Name, "m3") {
+				fmt.Println("An m3 pod was deleted:", pod.ObjectMeta.Name)
```

Comment:
we don't close the channel after a deletion.

Reply:
no, we wait instead for the creation, this delete print is for informative purposes :P